src/wavesurfer.js

```diff
@@ -92,6 +92,20 @@ var WaveSurfer = {
 
         // Click-to-seek
         this.drawer.on('click', function (e, progress) {
+            var nominalWidth = Math.round(
+                my.getDuration() * my.params.minPxPerSec * my.params.pixelRatio
+            );
+            var parentWidth = my.drawer.getWidth();
+
+            if (!my.params.fillParent && nominalWidth < parentWidth) {
+                var bbox = my.drawer.wrapper.getBoundingClientRect();
+                progress = (e.clientX - bbox.left) * my.params.pixelRatio / nominalWidth;
+
+                if (progress > 1) {
+                    progress = 1;
+                }
+            }
```

Comment:
This logic should be inserted here: https://github.com/katspaugh/wavesurfer.js/blob/master/src/drawer.js#L44 or https://github.com/katspaugh/wavesurfer.js/blob/master/src/drawer.js#L63, in the drawer code.